core/src/main/java/net/opentsdb/query/anomaly/BaseAnomalyNode.java

```diff
@@ -91,10 +118,13 @@ public BaseAnomalyNode(final QueryNodeFactory factory,
       }
     }
     if (ds == null) {
-      throw new IllegalStateException("Downsample can't be null.");
+      throw new IllegalStateException("Downsample can't be null for anomaly " +
+              "processing at this time.");
     }
-    final long query_time_span = context.query().endTime().msEpoch() - 
-        context.query().startTime().msEpoch();
+
+    dataSourceConfig = context.commonSourceConfig(this);
+    final long query_time_span = dataSourceConfig.endTimestamp().msEpoch() -
+            dataSourceConfig.startTimestamp().msEpoch();
```

Comment:
Why are we adding up start and end to get query time span ? Shouldnt it be end - start ?